### ISSUE TITLE
Saving to casted json column will store an escaped json string

### ISSUE THREAD

**Issue by adifaidz** [laravel/framework#20115]
- Laravel Version: 5.4.28
- PHP Version: 7.1.1
- Database Driver & Version: MySQL 5.7.11

### Description:
Saving model with json column will save an escaped json if cast. Am I missing something here? Is this intentional?

If not cast,
`[{"code": "a", "name": "a"}, {"code": "b", "name": "b"}, {"code": "c", "name": "c"}]`

If cast,
`"[{\"name\":\"a\",\"code\":\"a\"},{\"name\":\"b\",\"code\":\"b\"},{\"name\":\"c\",\"code\":\"c\"}]"`

### Steps To Reproduce:
Add to model 
``` php
    protected $casts = ['tags' => 'array'];
```
Then save the model

**Comment by themsaid** [comment]
Looks like something in your code is already serializing the attribute so when Eloquent serializes it again it appears like this, check your code as the current settings work just fine.

**Comment by adifaidz** [comment]
Thanks, I tried something just now and now I get what the cast attribute is really doing.

**Comment by ALTELMA** [comment]
@adifaidz How do  you fix it? still use model cast or not?

**Comment by ThunderBirdsX3** [comment]
@adifaidz When you store to DB, You should store in array format not `json_encode()`.

**Comment by ALTELMA** [comment]
@ThunderBirdsX3 If I do like that. I will get mysql error "Array to string conversion", Because I use model cast. It save array as json automically.

**Comment by ThunderBirdsX3** [comment]
@ALTELMA Look like your database version not support. Or you column not JSON type.

**Comment by ALTELMA** [comment]
My database support column json type. I can store directly, but cannot save from Laravel.

Bad Example:
`Insert into users ('name', 'meta') VALUES ('ALTELMA', {"facebook":"","google":"","linked":""})`

Should be:
`Insert into users ('name', 'meta') VALUES ('ALTELMA', '{"facebook":"","google":"","linked":""}')`

I don't know why laravel save value without quote.

**Comment by ThunderBirdsX3** [comment]
Show your Model and code.

**Comment by ALTELMA** [comment]
Model
```
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class User extends Model
{
    /**
     * The attributes that should be cast to native types.
     *
     * @var array
     */
    protected $casts = [
        'meta' => 'array',
    ];
```
Save
```
$user = new User;
foreach($inputs as $value) {
    $user->name = $value;
    $user->meta = $value;
}

$user->save();
```

**Comment by ThunderBirdsX3** [comment]
@ALTELMA `User::create(['name' => $inputs->name, 'meta' => $inputs->meta]);`
I'm not sure, What is your `$inputs` it's `$request`?

**Comment by ALTELMA** [comment]
Yes, it's $request, but why I cannot use save()?

**Comment by ThunderBirdsX3** [comment]
@ALTELMA What is your data in meta look like? `dd($request->meta);`

**Comment by ALTELMA** [comment]
Data store in array format like this.
["facebook":"123", "twitter":"456"]

Actually that's example data. Main point is when save data should be "{}" not {}.

I think I have problem only save() function.

**Comment by seandelaney** [comment]
> @adifaidz When you store to DB, You should store in array format not `json_encode()`.

This comment alone has saved my day. I had the same issue - casting JSON column and spent ~2 hours last night trying to fix it and couldn't. Following the advice helped me resolve it.

Thank you!

**Comment by NomisGnos** [comment]
For me, removing the $casts for the fields fixed my issue.  I suspect, like it was said before, that ->save() does something extra.

**Comment by benmAltDesign** [comment]
We found a way around this issue. 

We removed the json cast from the model ( We actually switched to our own model because it's not in the vendor folder but we copied all of the stuff from the existing DB model). This causes it to do weird stuff with the data but we then wrote these methods in

```php

public function getDataAttribute($value)
{
    return json_decode($value, true);
}

public function setDataAttribute($value)
{
    $this->attributes['data'] = json_encode($value, JSON_UNESCAPED_SLASHES);
}

```

These account for the removed casting and replace them with a load method and a save method that doesn't escape the JSON.
